extension/functions/src/postgres.rs

```diff
@@ -14,3 +14,149 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
+
+use arrow::array::*;
+// use arrow::array::{ArrayRef, BooleanArray, Int64Array,StringBuilder,Float64Array};
+// use arrow::array::{ StringArray,Array,};
+use arrow::array::{ArrayRef, TimestampNanosecondArray};
+use arrow::compute::cast;
+use arrow::datatypes::{self, IntervalUnit};
+use arrow::datatypes::{DataType, TimeUnit};
+use arrow::datatypes::{Field, IntervalDayTimeType};
+use datafusion::error::Result;
+// use datafusion::scalar::ScalarFunctionDef;
+//use datafusion::physical_plan::functions::{Signature, Volatility};
+use chrono::prelude::*;
+use chrono::Months;
+use chrono::{Datelike, Duration, NaiveDate, NaiveDateTime};
+use datafusion::logical_expr::Volatility;
+use datafusion::physical_plan::expressions::Column;
+use datafusion_common::cast::{
+    as_date32_array, as_float64_array, as_string_array, as_timestamp_nanosecond_array,
+};
+use datafusion_common::scalar::ScalarValue;
+use datafusion_common::DataFusionError;
+use datafusion_expr::ColumnarValue;
+use datafusion_expr::{
+    ReturnTypeFunction, ScalarFunctionDef, ScalarFunctionPackage, Signature,
+};
+use std::collections::HashSet;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct AgeFunction;
+
+impl ScalarFunctionDef for AgeFunction {
+    fn name(&self) -> &str {
+        "age"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::exact(
+            vec![
+                DataType::Timestamp(TimeUnit::Nanosecond, None),
```

Comment:
could also be second/millisecond/microsecond

Reply:
As long as test works, I think it is fine to only have nanosecond cc @sadboy

Reply:
The default is the microsecond. But looks like DF has type coercion from any timeunit to Nanosecond. So I think it is fine after I look at the tests